tareas/Tarea_2_Andres/intervalos.jl

```diff
@@ -0,0 +1,352 @@
+		else
+			@assert infimo ≤ supremo
+			return new{T}(infimo, supremo)
+		end
+	end
+
+# To allow that Intervalo(a) ≡ [a] → [a, a] also work
+	function Intervalo(same::T) where {T<:Real}
+		return Intervalo(same, same)
+	end
+end
+
+begin   ### Empty interval
+	function intervalo_vacio()   ### Without argument
+		return Intervalo(NaN)
+	end
+	function intervalo_vacio(arg::T) where {T<:Real}   ### A type Real 'number' arg
+		T₀ = promote_type(T, Float64)
+		return Intervalo(T₀(NaN), T₀(NaN))
+	end
+	function intervalo_vacio(T::Type{<:Real})   ### A 'type' arg
+		T₀ = promote_type(T, Float64)
+		return Intervalo(T₀(NaN), T₀(NaN))
+	end
+	function intervalo_vacio(I::Intervalo{T}) where {T<:Real}   ### An 'Interval' arg
+		return Intervalo(T(NaN), T(NaN))
+	end
+end
+
+begin   ### Checking if an interval I is the empty interval
+	import Base: isempty
+	function isempty(I::Intervalo)
+		return isnan(I.infimo)
+	end
+end
+
+begin   ### [a,b] == [c,d] ⇒ a = b && b = d
+	import Base: ==
+	function ==(I1::Intervalo, I2::Intervalo)
+		if isempty(I1) && isempty(I2)   ### Empty interval cases
+			return true
+		else
+			inf1, inf2, sup1, sup2 = promote(I1.infimo, I2.infimo, I1.supremo, I2.supremo)
+			return (inf1 == inf2) && (sup1 == sup2)
+		end
+	end
+end
+
+begin   ### [a,b] ⊆ [c,d] ⇒ c ≤ a && b ≤ d
+	import Base: ⊆
+	function ⊆(I1::Intervalo, I2::Intervalo)
+		if isempty(I1)   ### Empty I case
+			return true
+		elseif (I2.infimo ≤ I1.infimo) && (I1.supremo ≤ I2.supremo)
+			return true
+		else
+			return false
+		end
+	end
+end
+
+begin   ### [a,b] ⊂ [c,d] ⇒ c ≤ a && b ≤ d   so that   (c = a && b = d) == false
+	function ⊂(I1::Intervalo, I2::Intervalo)
+		if isempty(I1)   ### Empty I case
+			return true
+		elseif (I2.infimo ≤ I1.infimo) && (I1.supremo ≤ I2.supremo)
+			if (I2.infimo == I1.infimo) && (I1.supremo == I2.supremo)
+				return false
+			else
+				return true
+			end
+		else
+			return false
+		end
+	end
+	function ⊃(I1::Intervalo, I2::Intervalo)
+		if isempty(I2)   ### Empty I case
+			return true
+		elseif (I1.infimo ≤ I2.infimo) && (I2.supremo ≤ I1.supremo)
+			if (I1.infimo == I2.infimo) && (I2.supremo == I1.supremo)
+				return false
+			else
+				return true
+			end
+		else
+			return false
+		end
+	end
+end   ### Note: There were no methods for this symbols
+
+begin   ### [a,b] ⪽ [c,d] ⇒ c < a && b < d
+	function isinterior(I1::Intervalo, I2::Intervalo)
+		if isempty(I1)   ### Empty I case
+			return true
+		elseif (I2.infimo < I1.infimo) && (I1.supremo < I2.supremo)
+			return true
+		else
+			return false
+		end
+	end
+	const ⪽ = isinterior
+end
+
+begin   ### x ∈ [a,b] ⇒ a ≤ x ≤ b
+	import Base: ∈
+	function ∈(x::Real, I::Intervalo)
+		return (I.infimo ≤ x) && (x ≤ I.supremo)
+	end
+end
+
+begin   ### [a,b] ⊔ [c,d] = [min(a,c), max(b,d)]
+	function hull(I1::Intervalo, I2::Intervalo)
+		if isempty(I1)   ### Empty interval cases
+			return I2
+		elseif isempty(I2)
+			return I1
+		elseif isempty(I1) && isempty(I2)
+			return I1
```

Comment:
```suggestion
		elseif isempty(I2) || (isempty(I1) && isempty(I2))
			return I1
```

ya que en este caso existe más de una vez que se devuelve el mismo valor puedes ser conveniente hacer esto, pero ojo no siempre puede ser tan conveniente, abajo te doy un ejemplo.